prboom2/src/m_file.c

```diff
@@ -545,3 +550,46 @@ char *M_getenv(const char *name) {
   return getenv(name);
 #endif
 }
+
+static int nftwfunc(const char *filename, const struct stat *stats, int flags, struct FTW *pfwt)
+{
+  (void)stats;
+  (void)flags;
+  (void)pfwt;
+  return M_remove(filename);
+}
+
+#define NFTW_LIMIT 64
+
+dboolean M_RemoveDirRecursive(const char *path)
```

Comment:
This is a pretty dangerous function, so it might be good to add a check that `path` begins with a directory that DSDA has business modifying, does not contain anything that might escape (`..` components) and `I_Error` out otherwise.